code/DDSCodeTester.cpp

```diff
@@ -4486,11 +4554,837 @@ void log_examples()
+        // Set the two latest possible values on the sequence
+        sequence_data->set_int16_values(3, in_value);
+        // Read every sequence value from index 1 to the end
+        sequence_data->get_int16_values(out_value, 1);
+
+        int16_t in_simple_value = 8;
+        int16_t out_simple_value;
+        sequence_data->set_int16_value(2, in_simple_value);
+        sequence_data->get_int16_value(out_simple_value, 2);
+        
+        data->return_loaned_value(sequence_data);
+        //!--
+    }
+    {
+        //!--CPP_ARRAYS
+        // Define a struct type to contain the array
+        TypeDescriptor::_ref_type type_descriptor {traits<TypeDescriptor>::make_shared()};
+        type_descriptor->kind(TK_STRUCTURE);
+        type_descriptor->name("ArrayStruct");
+        DynamicTypeBuilder::_ref_type struct_builder {DynamicTypeBuilderFactory::get_instance()->
+                                                              create_type(type_descriptor)};
+        // Define a member for the array
+        MemberDescriptor::_ref_type array_member_descriptor {traits<MemberDescriptor>::make_shared()};
+        array_member_descriptor->name("long_array");
+        array_member_descriptor->type(DynamicTypeBuilderFactory::get_instance()->create_array_type(
+                    DynamicTypeBuilderFactory::get_instance()->get_primitive_type(TK_INT32), { 2, 3, 4 })->build());
+
+        /* Alternative
+            type_descriptor = traits<TypeDescriptor>::make_shared();
+            type_descriptor->kind(TK_ARRAY);
+            type_descriptor->type(DynamicTypeBuilderFactory::get_instance()->get_primitive_type(TK_INT32));
+            type_descriptor->bound().push_back(2);
+            type_descriptor->bound().push_back(3);
+            type_descriptor->bound().push_back(4);
+            array_member_descriptor->type(DynamicTypeBuilderFactory::get_instance()->create_type(
+                    type_descriptor)->build());
+        */
+
+        // Add the array member to the struct
+        struct_builder->add_member(array_member_descriptor);
+        // Build the struct type
+        DynamicType::_ref_type struct_type {struct_builder->build()};
+        // Create dynamic data based on the struct type
+        DynamicData::_ref_type data {DynamicDataFactory::get_instance()->create_data(struct_type)};
+
+        // Set and retrieve values for the array member
+        Int32Seq in_value = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 23, 23, 24};
+        Int32Seq out_value;
+        data->set_int32_values(data->get_member_id_by_name("long_array"), in_value);
+        data->get_int32_values(out_value, data->get_member_id_by_name("long_array"));
+
+        DynamicData::_ref_type array_data {data->loan_value(data->get_member_id_by_name("long_array"))};
+        // Set the two latest possible values on the array
+        array_data->set_int32_values(22, in_value);
```

Comment:
I think this will fail because you are passing a sequence with more than 2 elements.